pkg/module/security_scan.go

```diff
@@ -0,0 +1,162 @@
+package module
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	v2 "github.com/gardener/component-spec/bindings-go/apis/v2"
+	"github.com/gardener/component-spec/bindings-go/ctf"
+	"github.com/mandelsoft/vfs/pkg/vfs"
+	"os"
+	"path"
+	"sigs.k8s.io/yaml"
+	"strings"
+)
+
+var ErrFailedToParseImageURL = errors.New("error parsing protecode image URL")
+
+const (
+	secConfigFileName = "sec-scanners-config.yaml"
+	secScanLabelKey   = "scan.security.kyma-project.io"
+	secLabelKey       = "security.kyma-project.io"
+	secScanEnabled    = "enabled"
+)
+
+var labelTemplate = "%s." + secScanLabelKey
+var globalLabelTemplate = "%s." + secLabelKey
+
+func AddSecurityScanningMetadata(descriptor *v2.ComponentDescriptor, modDef *Definition, fs vfs.FileSystem,
+	securityConfigPath string,
+) error {
+	//parse security config file
+	config, err := parseSecurityScanConfig(securityConfigPath)
+	if err != nil {
+		return err
+	}
+	excludedWhitesourcePathPatterns := strings.Join(config.WhiteSource.Exclude, ",")
+
+	// add security scan enabled global label
+	err = appendLabelToAccessor(descriptor, "scan", secScanEnabled, globalLabelTemplate)
+	if err != nil {
+		return err
+	}
+	if len(descriptor.Sources) == 0 {
+		return errors.New("found no sources in component descriptor")
+	}
+	//add whitesource sec scan labels
+	for srcIdx := range descriptor.Sources {
+		src := &descriptor.Sources[srcIdx]
+		err := appendLabelToAccessor(src, "language", config.WhiteSource.Language, labelTemplate)
+		if err != nil {
+			return err
+		}
+		err = appendLabelToAccessor(src, "subprojects", config.WhiteSource.SubProjects, labelTemplate)
+		if err != nil {
+			return err
+		}
+		err = appendLabelToAccessor(src, "exclude", excludedWhitesourcePathPatterns, labelTemplate)
+		if err != nil {
+			return err
+		}
+	}
+	//add protecode sec scan images
+	if err := appendProtecodeImagesLayers(descriptor, config); err != nil {
+		return err
+	}
+
+	return WriteComponentDescriptor(fs, descriptor, modDef.ArchivePath,
+		ctf.ComponentDescriptorFileName)
+}
+
+func appendProtecodeImagesLayers(descriptor *v2.ComponentDescriptor, config *SecurityScanCfg) error {
+	for _, imageURL := range config.Protecode {
+		imageName, err := getImageName(imageURL)
+		if err != nil {
+			return err
+		}
+
+		imageTypeLabel, err := generateOCMLabel("type", "third-party-image", labelTemplate)
+		if err != nil {
+			return err
+		}
+		imageLayerMetadata := v2.IdentityObjectMeta{
+			Name:    imageName,
+			Type:    v2.OCIImageType,
+			Version: descriptor.Version,
+			Labels:  []v2.Label{imageTypeLabel},
+		}
+		imageLayerAccess, err := v2.NewUnstructured(v2.NewOCIRegistryAccess(imageURL))
```

Comment:
better mentioned in the [original issue](https://github.tools.sap/kyma/backlog/issues/3338) that the access layer format is different than proposed (there is no ref field), and also highlight it's according to gardener component-spec api definition https://github.com/gardener/component-spec/blob/master/bindings-go/apis/v2/accesstypes.go

```
  - access:
      imageReference: artifact.registry.hostname/path/to/the/registry/k8s-tools:v20221213-1840f45a
      type: ociRegistry
    labels:
    - name: type.scan.security.kyma-project.io
      value:
        type: third-party-image
    name: k8s-tools
    relation: external
    type: ociImage
    version: 0.3.4
```